code/game/objects/effects/anomalies/anomalies_ectoplasm.dm

```diff
@@ -0,0 +1,211 @@
+
+
+
+// Ghost Portal. Used to bring anomaly orbiters into the playing field as ghosts. Destroys itself and all of its associated ghosts after two minutes.
+// Can be destroyed early to the same effect.
+
+/obj/structure/ghost_portal
+	name = "Spooky Portal"
+	desc = "A portal between our dimension and who-knows-where? It's emitting an absolutely ungodly wailing sound."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "anom"
+	anchored = TRUE
+	var/static/list/spooky_noises = list(
+		'sound/hallucinations/growl1.ogg',
+		'sound/hallucinations/growl2.ogg',
+		'sound/hallucinations/growl3.ogg',
+		'sound/hallucinations/veryfar_noise.ogg',
+		'sound/hallucinations/wail.ogg'
+	)
+	var/list/ghosts_spawned = list()
+
+/obj/structure/ghost_portal/Initialize(mapload, candidate_list)
+	. = ..()
+
+	START_PROCESSING(SSobj, src)
+	INVOKE_ASYNC(src, PROC_REF(make_ghost_swarm), candidate_list)
+	playsound(src, pick(spooky_noises), 100, TRUE)
+	QDEL_IN(src, 2 MINUTES)
+
+/obj/structure/ghost_portal/process(delta_time)
+	. = ..()
+
+	if(prob(5))
+		playsound(src, pick(spooky_noises), 100)
+
+/obj/structure/ghost_portal/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	playsound(loc, 'sound/effects/empulse.ogg', 75, TRUE)
+	if(prob(40))
+		playsound(src, pick(spooky_noises), 50)
+
+/obj/structure/ghost_portal/Destroy()
+	. = ..()
+
+	STOP_PROCESSING(SSobj, src)
+	cleanup_ghosts()
+
+/**
+ * Generates a poll for observers, spawning anyone who signs up in a large group of ghost mobs
+ *
+ * Generates a poll that asks anyone observing for participation. Spawns a bunch of basicmob ghosts with the keys of candidates who have signed up.
+ * Ghosts are deleted two minutes after being made, and exist to punch stuff until it breaks.
+ */
+
+/obj/structure/ghost_portal/proc/make_ghost_swarm(list/candidate_list)
+	if(!length(candidate_list)) //If we are not passed a candidate list we just poll everyone who is dead, meaning these can also be spawned directly.
+		candidate_list += GLOB.current_observers_list
+		candidate_list += GLOB.dead_player_list
+
+	var/list/candidates = poll_candidates("Would you like to participate in a spooky ghost swarm? (Warning: you will not be able to return to your body!)", ROLE_SENTIENCE, FALSE, 10 SECONDS, group = candidate_list)
+	for(var/candidate in candidates)
+		if(!isobserver(candidate))
+			continue
+		var/mob/dead/observer/candidate_ghost = candidate
+		var/mob/living/basic/ghost/swarm/new_ghost = new(get_turf(src))
+		new_ghost.ghostize(FALSE)
+		new_ghost.key = candidate_ghost.key
+		new_ghost.log_message("was returned to the living world as a ghost by an ectoplasmic anomaly.", LOG_GAME)
+		var/policy = get_policy(ROLE_ANOMALY_GHOST)
+		if (policy)
+			to_chat(new_ghost, policy)
+		else
+			to_chat(new_ghost, span_revenboldnotice("You are a lost soul, brought back to the realm of the living. Your time on this plane is limited, and you will soon be dragged back into the void!"))
+		ghosts_spawned += new_ghost
```

Comment:
I would add this right after the line where we spawn the new ghost (187), just to be safe in case anything yorks and kills us and this ghost never gets cleaned up/accounted for ever